apps/dashboard/app/controllers/projects_controller.rb

```diff
@@ -11,7 +11,12 @@ def show
     else
       @scripts = Script.all(@project.directory)
       @valid_project = Script.clusters?
-      flash.now[:alert] = I18n.t("dashboard.jobs_project_invalid_configuration_clusters") unless @valid_project
+      @valid_scripts = Script.scripts?(@project.directory)
+
+      alert_messages = []
+      alert_messages << I18n.t('dashboard.jobs_project_invalid_configuration_clusters') unless @valid_project
+      alert_messages << I18n.t('dashboard.jobs_project_invalid_configuration_scripts') if @scripts.any? && !@valid_scripts
+      flash.now[:alert] = alert_messages.join(' ') if alert_messages.any?
```

Comment:
I cannot seem to get these alerts to show up. But honestly, if it creates a `hello_world.sh` automatically - I wonder if we need them at all?

I think creating the default script without asking/or alerting the user is actually a good first step. Maybe we could build in the alert/prompt later - but I'm now wondering what value it provides?

Maybe actually - just a note on the success notice like - 

![image](https://github.com/OSC/ondemand/assets/4874123/3b1e5aab-d99e-4930-878e-f2a40a8b62e6)

Reply:
The script error message will appear if there are Scripts created and afterwards, the user manually deletes all the executable scripts from the project. Then, when rendering the projects page, the error message will show.

As scripts are added/removed by the user with the file upload tool, it is possible that the user deletes all executables.

Reply:
> The script error message will appear if there are Scripts created and afterwards, the user manually deletes all the executable scripts from the project.

That was the step I was missing! Yes I'm now able to get the alert to flash. 

OK - While I'm fine for this for now, I would still make a ticket to make this `flash` a `modal` that you says something to the same effect, where you can click to create one.

```
An executable shell script is required for your project.

Would you like to create a hello world shell script?
[yes]   [no]
```

If you want to pick that up now or later, either is fine with me.

Reply:
I think later is better.

I am not sure about the modal as it will keep appearing and forcing the user to select an option every time the projects page is displayed (if they select no and until a script is created).

Reply:
OK.

> I am not sure about the modal as it will keep appearing and forcing the user to select an option every time the projects page is displayed (if they select no and until a script is created).

That's fine by me. They need a script for any of this to work, so it's either our modal telling them that or the `sbatch` or similar failure message when they attempt to start a job - which they may or may not understand.